conda_env/specs/binstar.py

```diff
@@ -0,0 +1,111 @@
+import re
+from conda.resolve import normalized_version
+from .. import env
+from ..exceptions import EnvironmentFileDoesNotExist, EnvironmentFileNotDownloaded, CondaEnvException
+try:
+    from binstar_client import errors
+    from binstar_client.utils import get_binstar
+except ImportError:
+    raise CondaEnvException("Binstar not installed")
+
+ENVIRONMENT_TYPE = 'env'
+
+
+class BinstarSpec(object):
```

Comment:
Not a hard requirement, but there's a lot in this object that I would refactor away into `conda_env.utils.binstar`.  All of the code that interacts with the results of binstar information should really be in the client library.  I should be able to do:

```python
pkg = binstar.Package("darth/deathstar")
if not pkg.has_environments:
    # handle error state
else:
    return env.from_yaml(pkg.environments.latest.raw.read())
```

If we can isolate the code into `conda_env.utils.binstar`, then we can adjust to that new API if/when it's available without having to touch this code.  It also makes it easier to pull that code into the binstar API if we want to just move it over there.

If you don't have the bandwidth right now for the change, do you mind putting in a `TODO`?